app/tests/app/models/variable.service.test.js

```diff
@@ -0,0 +1,368 @@
+        };
+      }
+    });
+  });
+
+  beforeEach(
+    inject(function(
+      _$resource_,
+      backendUrl,
+      _$http_,
+      _$cacheFactory_,
+      _$q_,
+      _Variable_,
+      _$httpBackend_
+    ) {
+      $resource = _$resource_;
+      $http = _$http_;
+      $cacheFactory = _$cacheFactory_;
+      $q = _$q_;
+      Variable = _Variable_;
+      $httpBackend = _$httpBackend_;
+    })
+  );
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it("Should return the whole hierarchical data when hierarchy() is called", function(
+    done
+  ) {
+    $httpBackend
+      .expectGET("http://localhost:8080/services/variables/hierarchy")
+      .respond(getMockHierarchy());
+
+    Variable.hierarchy().then(function(data) {
+      expect(data).toEqualData(getMockHierarchy());
+      done();
+    });
+
+    $httpBackend.flush();
+  });
+
+  it("Should return the variable name *Right Hippocampus* when getVariableData is called with the code *righthippocampus*", function(
+    done
+  ) {
+    $httpBackend
+      .expectGET("http://localhost:8080/services/variables/hierarchy")
+      .respond(getMockHierarchy());
+
+    Variable.getVariableData("righthippocampus").then(function(data) {
+      expect(data.data.label).toEqualData("Right Hippocampus");
+      done();
+    });
+
+    $httpBackend.flush();
+  });
+
+  it("Should return the Breadcrumb (or Path) to a variable (each node traversed from root) when getBreadcrumb is called with the code *righthippocampus*", function(
+    done
+  ) {
+    var target = [
+      { label: "/", code: "root", childsLength: 3 },
+      { label: "Brain Anatomy", code: "brain_anatomy", childsLength: 2 },
+      {
+        label: "Grey matter volume",
+        code: "grey_matter_volume",
+        childsLength: 3
+      },
+      { label: "Limbic", code: "limbic", childsLength: 0 },
+      { label: "Right Hippocampus", code: "righthippocampus", childsLength: 0 }
+    ];
+    $httpBackend
+      .expectGET("http://localhost:8080/services/variables/hierarchy")
+      .respond(getMockHierarchy());
+
+    Variable.getBreadcrumb("righthippocampus").then(function(data) {
+      expect(data).toEqualData(target);
+      done();
+    });
+
+    $httpBackend.flush();
+  });
+
+  it("Should return a collection of object detailing the code and variable count of sub categories when getSubCategoryVariableCounter is called with the code *brain_anatomy*", function(
+    done
+  ) {
+    var target = [
+      { code: "csf_volume", counter: 2 },
+      { code: "grey_matter_volume", counter: 9 }
+    ];
+
+    $httpBackend
+      .expectGET("http://localhost:8080/services/variables/hierarchy")
+      .respond(getMockHierarchy());
+
+    Variable.getSubCategoryVariableCounter("brain_anatomy").then(function(
+      data
+    ) {
+      expect(data).toEqualData(target);
+      done();
+    });
+
+    $httpBackend.flush();
+  });
+});
+
```

Comment:
function hoisting makes it convenient to store our mock data underneath the test code